GDJS/Runtime/inputmanager.ts

```diff
@@ -344,12 +363,22 @@ namespace gdjs {
       return this._startedTouches;
     }
 
+    /**
+     * @deprecated
+     */
     popStartedTouch(): integer | undefined {
-      return this._startedTouches.shift();
+      const touchId = this._startedTouches[this._lastStartedTouchIndex];
+      this._lastStartedTouchIndex++;
+      return touchId;
     }
 
+    /**
+     * @deprecated
+     */
     popEndedTouch(): integer | undefined {
-      return this._endedTouches.shift();
+      const touchId = this._endedTouches[this._lastEndedTouchIndex];
+      this._lastEndedTouchIndex++;
+      return touchId;
     }
```

Comment:
It concerns me that now, once an ID has been pushed into `_endedTouches` or `_startedTouches`, it will never get removed, so spam touching the screen will fill up memory slowly but steadily. I'd suggest clearing both arrays at the end of the frame since we usually only need to process the start or end of a touch once anyways and to keep the memory footprint low.

Reply:
It seems it was already the case, because nothing was forcing users to pop every events.

```JavaScript
    onFrameEnded(): void {
      //Only clear the ended touches at the end of the frame.
      for (const id in this._touches.items) {
        if (this._touches.items.hasOwnProperty(id)) {
          const touch = this._touches.items[id];
          if (touch.justEnded) {
            this._touches.remove(id);
          }
        }
      }
      this._startedTouches.length = 0;
      this._endedTouches.length = 0;
      this._releasedKeys.clear();
      this._releasedMouseButtons.length = 0;
      this._mouseWheelDelta = 0;
    }
```

Reply:
Ah, I missed that, all good then 👍

Reply:
(resolving)